Mathlib/NumberTheory/RamificationInertia/Galois.lean

```diff
@@ -0,0 +1,197 @@
+  maximal ideal `p` of `A` are the same, which we define as `Ideal.inertiaDegIn`. -/
+noncomputable def inertiaDegIn {A : Type*} [CommRing A] (p : Ideal A)
+    (B : Type*) [CommRing B] [Algebra A B] : ℕ :=
+  if h : ∃ P : Ideal B, P.IsPrime ∧ P.LiesOver p then p.inertiaDeg h.choose else 0
+
+section MulAction
+
+variable {A B : Type*} [CommRing A] [CommRing B] [Algebra A B] {p : Ideal A}
+
+instance : MulAction (B ≃ₐ[A] B) (primesOver p B) where
+  smul σ Q := primesOver.mk p (map σ Q.1)
+  one_smul Q := Subtype.val_inj.mp (map_id Q.1)
+  mul_smul σ τ Q := Subtype.val_inj.mp (Q.1.map_map τ.toRingHom σ.toRingHom).symm
+
+@[simp]
+theorem coe_smul_primesOver (σ : B ≃ₐ[A] B) (P : primesOver p B): (σ • P).1 = map σ P :=
+  rfl
+
+@[simp]
+theorem coe_smul_primesOver_mk (σ : B ≃ₐ[A] B) (P : Ideal B) [P.IsPrime] [P.LiesOver p] :
+    (σ • primesOver.mk p P).1 = map σ P :=
+  rfl
+
+variable (K L : Type*) [Field K] [Field L] [Algebra A K] [IsFractionRing A K] [Algebra B L]
+  [Algebra K L] [Algebra A L] [IsScalarTower A B L] [IsScalarTower A K L]
+  [IsIntegralClosure B A L] [FiniteDimensional K L]
+
+instance : MulAction (L ≃ₐ[K] L) (primesOver p B) where
+  smul σ Q := primesOver.mk p (map (galRestrict A K L B σ) Q.1)
+  one_smul Q := by
+    apply Subtype.val_inj.mp
+    show map _ Q.1 = Q.1
+    simpa only [map_one] using map_id Q.1
+  mul_smul σ τ Q := by
+    apply Subtype.val_inj.mp
+    show map _ Q.1 = map _ (map _ Q.1)
+    rw [_root_.map_mul]
+    exact (Q.1.map_map ((galRestrict A K L B) τ).toRingHom ((galRestrict A K L B) σ).toRingHom).symm
+
+theorem coe_smul_primesOver_eq_map_galRestrict (σ : L ≃ₐ[K] L) (P : primesOver p B):
+    (σ • P).1 = map (galRestrict A K L B σ) P :=
+  rfl
+
+theorem coe_smul_primesOver_mk_eq_map_galRestrict (σ : L ≃ₐ[K] L) (P : Ideal B) [P.IsPrime]
+    [P.LiesOver p] : (σ • primesOver.mk p P).1 = map (galRestrict A K L B σ) P :=
+  rfl
+
+end MulAction
+
+section RamificationInertia
+
+variable {A B : Type*} [CommRing A] [IsDomain A] [IsIntegrallyClosed A] [CommRing B] [IsDomain B]
+  [IsIntegrallyClosed B] [Algebra A B] [Module.Finite A B] (p : Ideal A) (P Q : Ideal B)
+  [hPp : P.IsPrime] [hp : P.LiesOver p] [hQp : Q.IsPrime] [Q.LiesOver p]
+  (K L : Type*) [Field K] [Field L] [Algebra A K] [IsFractionRing A K] [Algebra B L]
+  [IsFractionRing B L] [Algebra K L] [Algebra A L] [IsScalarTower A B L] [IsScalarTower A K L]
+  [FiniteDimensional K L]
+
+include p in
+/-- If `p` is a maximal ideal of `A`, `P` and `Q` are prime ideals
+  lying over `p`, then there exists `σ ∈ Aut (B / A)` such that `σ P = Q`. In other words,
+  the Galois group `Gal(L / K)` acts transitively on the set of all prime ideals lying over `p`. -/
+theorem exists_map_eq_of_isGalois [IsGalois K L] : ∃ σ : B ≃ₐ[A] B, map σ P = Q := by
+  haveI : NoZeroSMulDivisors A B := NoZeroSMulDivisors.of_field_isFractionRing A B K L
+  haveI : IsInvariant A B (B ≃ₐ[A] B) := isInvariant_of_isGalois' A K L B
```

Comment:
```suggestion
  have : NoZeroSMulDivisors A B := NoZeroSMulDivisors.of_field_isFractionRing A B K L
  have : IsInvariant A B (B ≃ₐ[A] B) := isInvariant_of_isGalois' A K L B
```
I don't think the `I` is necessary here (it stands for "inline", not "instance" as it used to be the case in Lean 3).